src/Analysis/Ast/Impl/Linting/UndefinedVariables/UndefinedVariablesWalker.cs

```diff
@@ -53,9 +53,13 @@ public UndefinedVariablesWalker(IDocumentAnalysis analysis, IServiceContainer se
                         augs.Right?.Walk(new ExpressionWalker(this));
                         break;
                     case AssignmentStatement asst:
+                        var lhs = asst.Left.MaybeEnumerate().ToArray();
+                        if (lhs.Length > 0) {
+                            lhs[0]?.Walk(new ExpressionWalker(this));
+                        }
                         _suppressDiagnostics = true;
-                        foreach (var lhs in asst.Left ?? Enumerable.Empty<Expression>()) {
-                            lhs?.Walk(new ExpressionWalker(this));
+                        foreach (var l in lhs.Skip(1)) {
```

Comment:
That parts works OK i.e. `x[y]` detects undefined `x` or `y`. The remaining part of LHS is trickier since we don't know if RHS returns proper number of items.

Reply:
I'll make a test for the case I'm thinking of and see if it works.